pkg/admiralctl/migrate/migrate.go

```diff
@@ -0,0 +1,694 @@
+	if err != nil {
+		return fmt.Errorf("failed to get gvr from %q: %w", o.gvk, err)
+	}
+
+	return o.migrate(controlPlaneRestConfig, obj, gvr)
+}
+
+func (o *CommandMigrateOption) migrate(controlPlaneRestConfig *rest.Config, obj *unstructured.Unstructured,
+	gvr schema.GroupVersionResource,
+) error {
+	if err := preprocessResource(obj); err != nil {
+		return fmt.Errorf("failed to preprocess resource %q(%s/%s) in control plane: %w", gvr, o.Namespace, o.name, err)
+	}
+
+	if o.OutputFormat != "" {
+		// only print the resource template and Policy
+		err := o.printObjectAndPolicy(obj)
+
+		return err
+	}
+
+	if o.DryRun {
+		return nil
+	}
+
+	controlPlaneDynamicClient := dynamic.NewForConfigOrDie(controlPlaneRestConfig)
+
+	kubeadmiralClient := fedclient.NewForConfigOrDie(controlPlaneRestConfig)
+
+	if len(obj.GetNamespace()) == 0 {
+		_, err := controlPlaneDynamicClient.Resource(gvr).Get(context.TODO(), o.name, metav1.GetOptions{})
+		if err == nil {
+			fmt.Printf("Resource %q(%s) already exist in Kubeadmiral control plane, "+
+				"you can edit PropagationPolicy and OverridePolicy to propagate it\n", gvr, o.name)
+			return nil
+		}
+
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get resource %q(%s) in control plane: %w", gvr, o.name, err)
+		}
+
+		_, err = controlPlaneDynamicClient.Resource(gvr).Create(context.TODO(), obj, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create resource %q(%s) in control plane: %w", gvr, o.name, err)
+		}
+
+		if o.AutoCreatePolicy {
+			policyName, err := o.createClusterPropagationPolicy(kubeadmiralClient)
+			if err != nil {
+				return err
+			}
+			patchData, err := getPatchData(obj, policyName)
+			if err != nil {
+				return err
+			}
+			_, err = controlPlaneDynamicClient.Resource(gvr).Namespace(o.Namespace).
+				Patch(context.TODO(), obj.GetName(), types.StrategicMergePatchType, patchData, metav1.PatchOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to update resource %q(%s/%s) in control plane: %w", gvr, obj.GetNamespace(), o.name, err)
+			}
+		}
+
+		fmt.Printf("Resource %q(%s) is migrated successfully\n", gvr, o.name)
+	} else {
+		_, err := controlPlaneDynamicClient.Resource(gvr).Namespace(o.Namespace).Get(context.TODO(), o.name, metav1.GetOptions{})
+		if err == nil {
+			fmt.Printf("Resource %q(%s/%s) already exist in Kubeadmiral control plane, "+
+				"you can edit PropagationPolicy and OverridePolicy to propagate it\n", gvr, o.Namespace, o.name)
+			return nil
+		}
+
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get resource %q(%s/%s) in control plane: %w", gvr, o.Namespace, o.name, err)
+		}
+
+		_, err = controlPlaneDynamicClient.Resource(gvr).Namespace(o.Namespace).Create(context.TODO(), obj, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create resource %q(%s/%s) in control plane: %w", gvr, o.Namespace, o.name, err)
+		}
+
+		if o.AutoCreatePolicy {
```

Comment:
Maybe we should support specifying exist propagation policy when federalize member cluster resources？ /cc @gary-lgy  @limhawjia